File: IntentExample/app/src/main/java/com/example/intentexample/MainActivity.java
package com.example.intentexample;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    private Button moveBtn;
    private EditText etText;
    private String str;
    private ImageView icBtn;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        moveBtn = findViewById(R.id.btn_move);
        etText = findViewById(R.id.et_text);
        icBtn = findViewById(R.id.icBtn);

        icBtn.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v){
                Toast.makeText(getApplicationContext(), "호호", Toast.LENGTH_SHORT).show();
            }
        });
        moveBtn.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v){
                str = String.valueOf(etText.getText());
                Intent intent = new Intent(MainActivity.this, SubActivity.class);
                intent.putExtra("str", str);
                startActivity(intent);  //액티비티 이동
            }
        });
    }
}

File: BMI_Calculator/app/src/main/java/fastcampus/BMI_Calculator/MainActivity.kt
package fastcampus.BMI_Calculator

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val heightEditText : EditText = findViewById(R.id.heightEditText)
        val weightEditText : EditText = findViewById(R.id.weightEditText)

        val resultBtn : Button = findViewById(R.id.resultBtn)


        resultBtn.setOnClickListener {
            if(heightEditText.text.isEmpty()|| weightEditText.text.isEmpty()){
                Toast.makeText(this, "빈 값이 있습니다", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }
            val height: Int = heightEditText.text.toString().toInt()
            val weight : Int = weightEditText.text.toString().toInt()

            val intent = Intent(this, ResultActivity::class.java)
            intent.putExtra("height",height)
            intent.putExtra("weight", weight)
            startActivity(intent)
        }


    }
}